lib/qr_code/cubit/qr_code_cubit.dart

```diff
@@ -0,0 +1,67 @@
+import 'dart:async';
+
+import 'package:bloc/bloc.dart';
+import 'package:domain_models/domain_models.dart';
+import 'package:equatable/equatable.dart';
+import 'package:intl/intl.dart';
+import 'package:qr_code_repository/qr_code_repository.dart';
+
+part 'qr_code_state.dart';
+
+class QrCodeCubit extends Cubit<QrCodeState> {
+  QrCodeCubit({
+    required QrCodeRepository qrCodeRepository,
+  })  : _qrCodeRepository = qrCodeRepository,
+        super(const QrCodeState()) {
+    fetchSeed();
+  }
+
+  final QrCodeRepository _qrCodeRepository;
+  StreamSubscription<int>? _tickerSubscription;
+
+  @override
+  Future<void> close() {
+    _tickerSubscription?.cancel();
+    return super.close();
+  }
+
+  Future<void> fetchSeed() async {
+    try {
+      final qrCode = await _qrCodeRepository.getSeed();
+
+      emit(
+        state.copyWith(
+          status: QrCodeStatus.success,
+          qrCode: qrCode,
+        ),
+      );
+
+      final dateExpiresAt = DateFormat('yyyy-MM-dd HH:mm:ss')
+          .parse(qrCode.expiresAt!.toString(), true);
+
+      await _tickerSubscription?.cancel();
+      _tickerSubscription = _qrCodeRepository
```

Comment:
Some questions here 🙂 

 - Why did you use a ticker instead of a periodic timer? 
 - Do you think that emitting a state on **each** new frame is a valuable solution? How would have a timer worked in this case?

Reply:
I've been use Ticker because I can reuse this in other parts of the app. It will be like a App State. So if I need to verify it in other widgets I should because the validation is happening in the repository.